### DISCUSSION TITLE
How do I use the NavigationView control? Is it a bug that directly assigns values to the menuitems property of a control in non mmvm mode, and the UI cannot display NavigationItem?

### DISCUSSION THREAD

**Discussion by BoyFaceGirl** [lepoco/wpfui#716] [Q&A]
c#  code-behind:

 public partial class MainWindow : Window
    {

        public ObservableCollection<object> _MenuItems;

        public MainWindow()
        {
            InitializeComponent();

            this.Loaded += MainWindow_Loaded;

            this._MenuItems = new ObservableCollection<object>
            {
                new NavigationViewItem()
                {
                    Content="游戏",
                    Icon = new SymbolIcon { Symbol = SymbolRegular.CheckboxChecked24 },
                    Foreground=Brushes.Black,
                    Background=Brushes.White,
                    MenuItems = new object[]
                    {
                        new NavigationViewItem()
                        {
                              Content="魔兽争霸",
                               Foreground=Brushes.Black,
                              Icon = new SymbolIcon { Symbol = SymbolRegular.CheckboxChecked24 },
                        }
                    }
                }
            };

            this.RootNavigation.MenuItems = _MenuItems;

        }

xaml code:

            <ui:NavigationView x:Name="RootNavigation" 
                               Grid.Row="1" 
                               IsBackButtonVisible="Collapsed"
                               Padding="42,0,42,0"
                               OpenPaneLength="310">           

            </ui:NavigationView>

![image](https://github.com/lepoco/wpfui/assets/8773178/8de21a0f-ab60-4dd6-82ef-a4b9bda7b4e7)

**Comment by FQMWakeZero** [comment]
这是我的导航栏代码，你可以参考一下：

Model Code Fragment :
```
...
        [ObservableProperty]
        private ObservableCollection<object> _menuItems = new()
        {
            new NavigationViewItem()
            {
                Content = "模块",
                Icon = new SymbolIcon { Symbol = SymbolRegular.Calendar3Day24 },
                TargetPageType = typeof(AssemblyPage),
            },
            new NavigationViewItem()
            {
                Content = "组件",
                Icon = new SymbolIcon { Symbol = SymbolRegular.CalendarWeekNumbers24 },
                TargetPageType = typeof(Views.Pages.Samples.MainUI_2),
                
            },
        };
...
```

Page XML :
```
<Page x:Name="page"
    x:Class="RobotizationGUI.Views.Pages.DashboardPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:local="clr-namespace:RobotizationGUI.Views.Pages"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:ui="http://schemas.lepo.co/wpfui/2022/xaml"
    xmlns:samples="clr-namespace:RobotizationGUI.Views.Pages"
    Title="DashboardPage"
    d:DataContext="{d:DesignInstance local:DashboardPage,
                                     IsDesignTimeCreatable=False}"
    
    d:DesignHeight="450"
    d:DesignWidth="800"
    ui:Design.Background="{DynamicResource ApplicationBackgroundBrush}"
    ui:Design.Foreground="{DynamicResource TextFillColorPrimaryBrush}"
    Foreground="{DynamicResource TextFillColorPrimaryBrush}"
    mc:Ignorable="d" ScrollViewer.VerticalScrollBarVisibility="Disabled">

    <Grid>
        <ui:NavigationView
            x:Name="NavigationView"
            FooterMenuItemsSource="{Binding ViewModel.FooterMenuItems}"
            MenuItemsSource="{Binding ViewModel.MenuItems}"
            PaneDisplayMode="LeftFluent" >
        </ui:NavigationView>
    </Grid>
</Page>

```

希望这些代码可以帮助到你，这是部分截屏内容
![image](https://github.com/lepoco/wpfui/assets/82809023/34e97541-6c9d-4511-a5ee-11f40fa6c0b4)

**Comment by BoyFaceGirl** [comment]
> 这是我的导航栏代码，你可以参考一下：
> 
> Model Code Fragment :
> 
> ```
> ...
>         [ObservableProperty]
>         private ObservableCollection<object> _menuItems = new()
>         {
>             new NavigationViewItem()
>             {
>                 Content = "模块",
>                 Icon = new SymbolIcon { Symbol = SymbolRegular.Calendar3Day24 },
>                 TargetPageType = typeof(AssemblyPage),
>             },
>             new NavigationViewItem()
>             {
>                 Content = "组件",
>                 Icon = new SymbolIcon { Symbol = SymbolRegular.CalendarWeekNumbers24 },
>                 TargetPageType = typeof(Views.Pages.Samples.MainUI_2),
>                 
>             },
>         };
> ...
> ```
> 
> Page XML :
> 
> ```
> <Page x:Name="page"
>     x:Class="RobotizationGUI.Views.Pages.DashboardPage"
>     xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
>     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
>     xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
>     xmlns:local="clr-namespace:RobotizationGUI.Views.Pages"
>     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
>     xmlns:ui="http://schemas.lepo.co/wpfui/2022/xaml"
>     xmlns:samples="clr-namespace:RobotizationGUI.Views.Pages"
>     Title="DashboardPage"
>     d:DataContext="{d:DesignInstance local:DashboardPage,
>                                      IsDesignTimeCreatable=False}"
>     
>     d:DesignHeight="450"
>     d:DesignWidth="800"
>     ui:Design.Background="{DynamicResource ApplicationBackgroundBrush}"
>     ui:Design.Foreground="{DynamicResource TextFillColorPrimaryBrush}"
>     Foreground="{DynamicResource TextFillColorPrimaryBrush}"
>     mc:Ignorable="d" ScrollViewer.VerticalScrollBarVisibility="Disabled">
> 
>     <Grid>
>         <ui:NavigationView
>             x:Name="NavigationView"
>             FooterMenuItemsSource="{Binding ViewModel.FooterMenuItems}"
>             MenuItemsSource="{Binding ViewModel.MenuItems}"
>             PaneDisplayMode="LeftFluent" >
>         </ui:NavigationView>
>     </Grid>
> </Page>
> ```
> 
> 希望这些代码可以帮助到你，这是部分截屏内容 ![image](https://user-images.githubusercontent.com/82809023/261587225-34e97541-6c9d-4511-a5ee-11f40fa6c0b4.png)

  I don't like using the mvvm method. The mvvm mode is too cumbersome and cumbersome, requiring a large number of viemodels to be written, and processing the ui is not user-friendly enough
